src/mrdsrg-spin-adapted/sadsrg.cc

```diff
@@ -393,11 +364,13 @@ double SADSRG::compute_reference_energy(BlockedTensor H, BlockedTensor F, Blocke
     /// E = 0.5 * ( H["ij"] + F["ij"] ) * L1["ji"] + 0.5 * V["xyuv"] * L2["uvxy"]
     /// Note that L1_mn = 2.0 * δ_mn now
 
+    size_t ncore = core_mos_.size();
     double E = Efrzc_ + Enuc_;
 
-    for (size_t m = 0, nc = core_mos_.size(); m < nc; ++m) {
-        E += H.block("cc").data()[m * nc + m];
-        E += F.block("cc").data()[m * nc + m];
+#pragma omp parallel for reduction(+ : E)
+    for (size_t m = 0; m < ncore; ++m) {
+        E += H.block("cc").data()[m * ncore + m];
+        E += F.block("cc").data()[m * ncore + m];
```

Comment:
Here I don't think that omp really helps us since we pick at most < 10000 elements. What makes sense to avoid is getting the pointer to the data via the data() function, which could be a significant overhead. So I would do instead:
```suggestion
    auto &H_cc = H.block("cc").data();
    auto &F_cc = F.block("cc").data();    
    for (size_t m = 0; m < ncore; ++m) {
        E += H_cc[m * ncore + m];
        E += F_cc[m * ncore + m];
```

Reply:
Thank you for the suggestion. I have made the changes.